test/TestProjects/Models-Cadl/Models-Cadl.cadl

```diff
@@ -0,0 +1,101 @@
+import "@cadl-lang/rest";
+
+@serviceTitle("Models in CADL")
+@serviceVersion("0.1.0")
+@doc("CADL project to test various types of models.")
+namespace ModelsInCadl;
+
+using Cadl.Http;
+using Cadl.Rest;
+
+@doc("Base model")
+model BaseModel {
+
+}
+
+@doc("Derived model")
+model DerivedModel extends BaseModel {
+  @doc("Required collection")
+  requiredCollection: CollectionItem[];
+}
+
+@discriminator("discriminatorProperty")
+@doc("Base model with discriminator property.")
+model BaseModelWithDiscriminator extends BaseModel {}
```

Comment:
discriminator is an property of the BaseModelWithDiscriminator, why is it not defined in the model definition?

Reply:
That is the way it is implemented in CADL: https://github.com/microsoft/cadl/blob/0a528079a54ab74de2614cbdd59eb4a907d58e05/docs/tutorial.md#polymorphism-with-discriminators